pytorch_lightning/trainer/trainer.py

```diff
@@ -153,7 +153,15 @@ def __init__(
         terminate_on_nan: bool = False,
         auto_scale_batch_size: Union[str, bool] = False,
         prepare_data_per_node: Optional[bool] = None,
-        plugins: Optional[Union[List[Union[Plugin, ClusterEnvironment, str]], Plugin, ClusterEnvironment, str]] = None,
+        plugins: Optional[
+            Union[
+                List[Union[TrainingTypePlugin, PrecisionPlugin, ClusterEnvironment, str]],
+                TrainingTypePlugin,
+                PrecisionPlugin,
+                ClusterEnvironment,
+                str,
+            ]
+        ] = None,
```

Comment:
let's define a placeholder in utilities/types.py ?

Reply:
TrainingTypepluin is importing types from utilities/types already, I think define this in types.py will cause circular import

Reply:
It would trigger a circular import if utilities/types was importing TrainingTypeplugin right ? It won't be the case here or I am missing something ?

Reply:
It's true that `types.py` can't import our code otherwise it's circular hell.

I've added the utility to `plugins/__init__.py` instead